src/common/buffer.cc

```diff
@@ -1523,16 +1523,15 @@ static ceph::spinlock debug_lock;
    */
   char *buffer::list::c_str()
   {
-    switch (get_num_buffers()) {
-    case 0:
-      // no buffers
-      return nullptr;
-    case 1:
-      // good, we're already contiguous.
-      break;
-    default:
+    if (length() == 0) {
+      return nullptr;                         // no non-empty buffers
+    }
+
+    auto iter = std::cbegin(_buffers);
+    ++iter;
+
+    if (iter != std::cend(_buffers)) {
```

Comment:
for the special case a solely empty bptr, it return immediately, but for other normal cases, there are  more code to execute( get iterator and compare iterator etc),  I am not sure if it is worth it.

Reply:
These iterators, in the opposition to the `bufferlist::iterator`, are nothing more than wrapped pointers.